File: snippets/js/src/req/fetch.ts
// ANCHOR: full
import {Client, Filter, Kind, KindStandard, Duration, loadWasmAsync} from "@rust-nostr/nostr-sdk";

async function fetch() {
    await loadWasmAsync();

    // ANCHOR: client
    let client = new Client();

    await client.addRelay("wss://relay.damus.io")
    await client.connect();
    // ANCHOR_END: client

    // ANCHOR: fetch
    let filter1 = new Filter().kind(Kind.fromStd(KindStandard.Metadata)).limit(3);
    let events1 = await client.fetchEvents(filter1, Duration.fromSecs(10));
    // ANCHOR_END: fetch

    // ANCHOR: fetch-from
    let filter2 = new Filter().kind(Kind.fromStd(KindStandard.TextNote)).limit(5);
    let events2 = await client.fetchEventsFrom(["wss://relay.damus.io"], filter2, Duration.fromSecs(10));
    // ANCHOR_END: fetch-from
}

fetch();
// ANCHOR_END: full

File: snippets/kotlin/src/main/kotlin/req/Fetch.kt
package rust.nostr.snippets.req

// ANCHOR: full
import java.time.Duration
import kotlinx.coroutines.runBlocking
import rust.nostr.sdk.*

suspend fun fetch() {
    // ANCHOR: client
    val client = Client()

    client.addRelay("wss://relay.damus.io")
    client.connect()
    // ANCHOR_END: client

    // ANCHOR: fetch
    val filter1: Filter = Filter().kind(Kind.fromStd(KindStandard.METADATA)).limit(3u)
    val events1: Events = client.fetchEvents(filter = filter1, timeout = Duration.ofSeconds(10L))
    // ANCHOR_END: fetch

    // ANCHOR: fetch-from
    val filter2: Filter = Filter().kind(Kind.fromStd(KindStandard.TEXT_NOTE)).limit(5u)
    val events2: Events = client.fetchEventsFrom(
        urls = listOf("wss://relay.damus.io"),
        filter = filter2,
        timeout = Duration.ofSeconds(10L)
    )
    // ANCHOR_END: fetch-from
}

fun main() {
    runBlocking { fetch() }
}
// ANCHOR_END: full